test/test_cuda.py

```diff
@@ -803,6 +803,24 @@ def test_load_nonexistent_device(self):
         with self.assertRaisesRegex(RuntimeError, msg):
             _ = torch.load(buf)
 
+    def test_specify_improper_device_name(self):
+        import os
+        with self.assertRaisesRegex(RuntimeError, "Expected one of cpu"):
+            fname = "tempfile.pt"
+            torch.save([torch.nn.Parameter(torch.randn(10, 10))], fname,
+                       _use_new_zipfile_serialization=True)
+            torch.load(fname, 'cuda0')
+        os.remove(fname)
```

Comment:
nit: it's more robust to do something like try...finally. And this is kind of risky thing to do, since you don't know if someone else is going to clobber this file. I'd kind of have liked to have seen the old code using tempfile; I might be abble to say what the bug would have been in that case.

Reply:
Alright I can put it in a try catch block. This comment has reasons behind the failure: https://github.com/pytorch/pytorch/pull/29087#issuecomment-563678888

Reply:
The comment doesn't tell me much. Do you have a copy of the old code somewhere, perhaps in your reflog?